### ISSUE TITLE
Find suitable replacement for forked github.com/robfig/cron.

### ISSUE THREAD

**Issue by oklahomer** [oklahomer/go-sarah#48]
Instead of simply employing original [github.com/robfig/cron](https://github.com/robfig/cron), this project uses [forked and customized version](https://github.com/oklahomer/cron) of this. The customization was required to assign identifier for each registered cron job to remove or replace on [live configuration update](https://github.com/oklahomer/go-sarah/wiki/Live-Configuration-Update). This customization however lowers maintenancibility, and hence a suitable replacement with well-maintained equivalent package is **want**ed.

The replacing package **must** have abilities to:
- execute registered jobs in a scheduled manner
- execute jobs in a panic-proof manner
- register a job with pre-declared identifier or dispense one on schedule registration
- remove registered scheduled job by above identifier
- cancel all scheduled jobs by context cancellation or by calling one method
- parse crontab-styled schedule declaration

When two or more packages meet above requirements and have equal popularity, the one with minimal features and simple implementation **should** have higher priority.

**Comment by oklahomer** [comment]
A newer version, [`v3`](https://godoc.org/gopkg.in/robfig/cron.v3) is currently available. [Its author](https://github.com/robfig/cron/issues/189) confirmed that v3 contains features of v1 and v2 along with all outstanding issue fixes and feature requests. This migration task can be included to go-sarah's `v2` development.

**Comment by oklahomer** [comment]
The module is successfully added to go.mod by running `go get`, but some workaround was required.
First I referred to https://gopkg.in/robfig/cron.v3 and tried with `go get gopkg.in/robfig/cron.v3` as below:
```
> go get gopkg.in/robfig/cron.v3
go: gopkg.in/robfig/cron.v3@v3.0.0-rc1: go.mod has non-....v3 module path "github.com/robfig/cron/v3" at revision v3.0.0-rc1
go: error loading module requirements
```

To avoid such error, the next trial used a commit hash hoping that [psudo-versions](https://golang.org/cmd/go/#hdr-Pseudo_versions) was automatically detected and generated by `go get` command.
> Pseudo-versions never need to be typed by hand: the go command will accept the plain commit hash and translate it into a pseudo-version (or a tagged version if available) automatically. This conversion is an example of a module query.
```
> go get github.com/robfig/cron@0f01e6b17                                                                                                                                                                                                         go: finding github.com/robfig/cron 0f01e6b17
go: github.com/robfig/cron@v0.0.0-20190503020742-0f01e6b177f4: go.mod has post-v0 module path "github.com/robfig/cron/v3" at revision 0f01e6b177f4
go: error loading module requirements
```
Error saying 'go.mod has post-v0 module path "**github.com/robfig/cron/v3**"' was returned.

With the above error message, it looked like the correct way was to designate the package name as "github.com/robfig/cron/v3."
```
> go get github.com/robfig/cron/v3@0f01e6b17                                                                                                                                                                                                      
go: finding github.com/robfig/cron/v3 0f01e6b17
go: downloading github.com/robfig/cron/v3 v3.0.0-rc1
go: extracting github.com/robfig/cron/v3 v3.0.0-rc1
```
Finally running `go get github.com/robfig/cron/v3@0f01e6b17` succeeded.

**Comment by oklahomer** [comment]
CI introduced some different issues with different Golang versions.

# Issues

## Semantic Import Versioning
Apparently, CI fails with Golang v1.10 or older because of [Semantic Import Versioning](https://github.com/golang/go/wiki/Modules#semantic-import-versioning).

> - If the module is version v2 or higher, the major version of the module must be included as a /vN at the end of the module paths used in go.mod files (e.g., module github.com/my/mod/v2, require github.com/my/mod/v2 v2.0.0) and in the package import path (e.g., import "github.com/my/mod/v2/mypkg").
> - If the module is version v0 or v1, do not include the major version in either the module path or the import path.

The import path includes `v3` at the end as below, which makes Golang 1.10 or older unrecognizable.
https://github.com/oklahomer/go-sarah/blob/29efe8615bb0f7ca5270bbc88d41f13a0f4fbed3/scheduler.go#L6
So the best way is to let `go get gopkg.in/robfig/cron.v3` work since any Golan version can identify such an import path.

## Supported Versions
The most important and critical issue with v3 at this point is that this version only supports Golang 1.12.0 or later.
https://github.com/robfig/cron/blob/0f01e6b177f4bd77ee0d887fe74ac66da30a6e00/go.mod#L3

# Solution
## Use v2 instead of v3
v2 does not support a feature to optionally supply location information to switch time zone.
https://github.com/oklahomer/go-sarah/blob/29efe8615bb0f7ca5270bbc88d41f13a0f4fbed3/scheduler.go#L59

## Use github.com/oklahomer/cron
Wait until...
- Golang 1.12 becomes the minimal version to support
- Or v3 of github.com/cron/robfig officially supports Golang 1.11

Support for Golang 1.10 and older version can be removed from `go-sarah`'s v2 release. Golang 1.13 release is around the corner and when it is released, support for two older major versions (1.11 and 1.12) should be enough.

**Comment by oklahomer** [comment]
Golang 1.11.0-1.11.3 fail to run `go get` when the depending module has `go.mod` and its `go` directive has `go 1.12` or higher. Using Golang 1.11.4 or higher solves this.
